x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -459,7 +460,10 @@ synchronized void markAsFailed(String reason, ActionListener<Void> listener) {
         persistStateToClusterState(newState, ActionListener.wrap(
             r -> listener.onResponse(null),
             e -> {
-                logger.error(new ParameterizedMessage("[{}] failed to set task state as failed to cluster state.", getTransformId()),
+                String msg = "Failed to persist to cluster state while marking task as failed with reason [" + reason + "].";
```

Comment:
Unrelated: In this function `markAsFailed we audit a message if the DF is failed while it is stopping (L432). That is definitely a log-able event as it aids understanding but I'm not sure it should be audited.